tests/models/test_gpg.py

```diff
@@ -0,0 +1,163 @@
+
+
+@unittest.skipIf(not have_gpg(), "gpg not found")
+class TestLegacyGPGKeyAndSigner(unittest.TestCase, TmpDirMixin, GPGKeysMixin):
+  """Test RSA gpg signature creation and verification."""
+
+  @classmethod
+  def setUpClass(cls):
+    cls.set_up_test_dir()
+    cls.set_up_gpg_keys()
+
+    cls.test_data = b"test_data"
+    cls.wrong_data = b"something malicious"
+
+    cls.default_keyid = cls.gpg_key_0C8A17
+    cls.signing_subkey_keyid = cls.gpg_key_D924E9
+    cls.expired_keyid = "e8ac80c924116dabb51d4b987cb07d6d2c199c7c"
+
+    cls.default_key_dict = export_pubkey(cls.default_keyid, cls.gnupg_home)
+
+  @classmethod
+  def tearDownClass(cls):
+    cls.tear_down_test_dir()
+
+  def test_gpg_sign_and_verify_object_with_default_key(self):
+    """Create and verify a signature using the default key on the keyring."""
+
+    # Create a signature.
+    signer = _LegacyGPGSigner(homedir=self.gnupg_home)
+    signature = signer.sign(self.test_data)
+
+    # Generate Key from gnupg keyring.
+    key = _LegacyGPGKey.from_keyring(self.default_keyid, self.gnupg_home)
+
+    key.verify_signature(signature, self.test_data)
+    with self.assertRaises(UnverifiedSignatureError):
+      key.verify_signature(signature, self.wrong_data)
+
+    # Generate Key from dict.
+    key = _LegacyGPGKey.from_legacy_dict(self.default_key_dict)
+
+    key.verify_signature(signature, self.test_data)
+    with self.assertRaises(UnverifiedSignatureError):
+      key.verify_signature(signature, self.wrong_data)
+
+  def test_gpg_sign_and_verify_object(self):
+    """Create and verify a signature using the specific key on the keyring."""
+
+    # Create a signature.
+    signer = _LegacyGPGSigner(self.signing_subkey_keyid, self.gnupg_home)
+    signature = signer.sign(self.test_data)
+
+    # Generate Key from gnupg keyring.
+    key = _LegacyGPGKey.from_keyring(self.signing_subkey_keyid, self.gnupg_home)
+
+    key.verify_signature(signature, self.test_data)
+    with self.assertRaises(UnverifiedSignatureError):
+      key.verify_signature(signature, self.wrong_data)
+
+    # Generate Key from dict.
+    key_dict = export_pubkey(self.signing_subkey_keyid, self.gnupg_home)
+    key = _LegacyGPGKey.from_dict(key_dict["keyid"], key_dict)
+
+    key.verify_signature(signature, self.test_data)
+    with self.assertRaises(UnverifiedSignatureError):
+      key.verify_signature(signature, self.wrong_data)
+
+  def test_verify_using_expired_keyid(self):
+    """Creates and verifies a signature using expired key on the keyring."""
+
+    # Create a signature.
+    signer = _LegacyGPGSigner(self.signing_subkey_keyid, self.gnupg_home)
+    signature = signer.sign(self.test_data)
+
+    # Verify signature using expired key.
+    key = _LegacyGPGKey.from_keyring(self.expired_keyid, self.gnupg_home)
+    with self.assertRaises(VerificationError):
+      key.verify_signature(signature, self.test_data)
+
+  def test_gpg_signer_serialization(self):
```

Comment:
Minor nit: This should be 
```suggestion
  def test_gpg_signature_serialization(self):
```
Right?